plugins/inputs/suricata/suricata.go

```diff
@@ -215,6 +227,28 @@ func (s *Suricata) parse(acc telegraf.Accumulator, sjson []byte) {
 	}
 }
 
+func (s *Suricata) parse(acc telegraf.Accumulator, sjson []byte) {
+	// initial parsing
+	var result map[string]interface{}
+	err := json.Unmarshal(sjson, &result)
+	if err != nil {
+		acc.AddError(err)
+		return
+	}
```

Comment:
How about actually propagating the error upward?

Reply:
Great!